collect_app/src/test/java/org/odk/collect/android/projects/AutomaticProjectCreatorDialogTest.kt

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.projects
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.notNullValue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.android.fragments.support.DialogFragmentHelpers
+import org.odk.collect.android.support.CollectHelpers
+import org.odk.collect.android.support.TestScreenContextActivity
+import org.odk.collect.android.utilities.DialogUtils
+import org.odk.collect.testshared.RobolectricHelpers
+
+@RunWith(AndroidJUnit4::class)
+class AutomaticProjectCreatorDialogTest {
```

Comment:
This test doesn't use `FragmentScenario` like `ManualProjectCreatorDialogTest` because then something doesn't work. From what I've noticed it's because of the fact that the dialog needs activity https://github.com/getodk/collect/pull/4591/files#diff-78a8fdb05c2a21720fa00d2097e76a82f212c91fa10693d078bd6c9338b0c601R131 and somehow the one provided 

and somehow the one provided by `FragmentScenario` had problems with that.

Reply:
Actually extracting the barcode setup stuff (like I suggest in another comment) might allow you to then stub it here so that you don't have to worry about the barcode integration in tests.